diff --git a/buildSrc/src/main/kotlin/Helpers.kt b/buildSrc/src/main/kotlin/Helpers.kt
index 4230e1c..7df1ab4 100644
--- a/buildSrc/src/main/kotlin/Helpers.kt
+++ b/buildSrc/src/main/kotlin/Helpers.kt
@@ -336,7 +336,7 @@ fun Project.setupPlugin(projectName: String) {
         }
 
         splits.abi {
-            isEnable = true
+            isEnable = false
             isUniversalApk = false
 
             if (targetAbi.isNotBlank()) {
@@ -348,6 +348,7 @@ fun Project.setupPlugin(projectName: String) {
         flavorDimensions("vendor")
         productFlavors {
             create("oss")
+            create("fdroid")
             create("fdroidArm64") {
                 versionNameSuffix = "-arm64"
             }
@@ -454,7 +455,7 @@ fun Project.setupApp() {
         }
 
         splits.abi {
-            isEnable = true
+            isEnable = false
             isUniversalApk = false
 
             if (targetAbi.isNotBlank()) {
@@ -467,6 +468,7 @@ fun Project.setupApp() {
         productFlavors {
             create("oss")
             create("expert")
+            create("fdroid")
             create("fdroidArm64") {
                 versionNameSuffix = "-arm64"
             }
@@ -523,4 +525,4 @@ fun Project.setupApp() {
     }
 
     setupPlay()
-}
\ No newline at end of file
+}
