diff --git a/buildSrc/src/main/kotlin/Helpers.kt b/buildSrc/src/main/kotlin/Helpers.kt
index 440dbdf..b950f35 100644
--- a/buildSrc/src/main/kotlin/Helpers.kt
+++ b/buildSrc/src/main/kotlin/Helpers.kt
@@ -335,7 +335,7 @@ fun Project.setupPlugin(projectName: String) {
         }
 
         splits.abi {
-            isEnable = true
+            isEnable = false
             isUniversalApk = false
 
             if (targetAbi.isNotBlank()) {
@@ -347,6 +347,7 @@ fun Project.setupPlugin(projectName: String) {
         flavorDimensions("vendor")
         productFlavors {
             create("oss")
+            create("fdroid")
             create("fdroidArm64") {
                 versionNameSuffix = "-arm64"
             }
@@ -453,7 +454,7 @@ fun Project.setupApp() {
         }
             create("fdroidArm64") {
                 versionNameSuffix = "-arm64"
             }
@@ -525,4 +527,4 @@ fun Project.setupApp() {
     }
 
     setupPlay()
-}
\ No newline at end of file
+}
