diff --git a/lib/main.dart b/lib/main.dart
index e1af2fc1..1372f84b 100644
--- a/lib/main.dart
+++ b/lib/main.dart
@@ -620,28 +620,6 @@ class _MyAppState extends State<MyApp> {
                       12) {
             return;
           }
-          Map<String, dynamic> message = {
-            // Use a fixed UserId for debugging
-            "UserId": kDebugMode
-                ? "f16dfa0a-f7b2-4f13-bb33-676e09788819"
-                : prefs.getString("userId")!,
-            "Timestamp": timeNowString,
-            "deviceInfo": await getDeviceInfo(),
-            "language": language,
-            "romanization": romanization,
-            "egJumpyPrs": egJumpyPrs,
-            "numBookmarks": numBookmarks,
-            "numHistory": numHistory,
-            "entryEgPrMethod": entryEgPrMethod,
-            "entryEgFontSize": entryEgFontSize,
-            ...analyticsState.toJson(),
-          };
-          final ok = await awsService.sendMessage(jsonEncode(message));
-          if (ok) {
-            // Clear analytics state if successfully sent
-            analyticsState.clear();
-            prefs.setString("analyticsSyncTime", timeNowString);
-          }
         }
       },
       child: Portal(
diff --git a/pubspec.yaml b/pubspec.yaml
index 974c6471..3486bd3c 100644
--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -83,7 +83,6 @@ flutter:
     - assets/
     - assets/images/
     - assets/jyutping_female/
-    - aws_secrets.json
   fonts:
     - family: ChironHeiHK
       fonts:
