diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 712ace14..9bd9d0df 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -255,17 +255,17 @@ if( ANDROID )
     add_compile_definitions(MANUAL_LOCATION="/android_asset/manual")
 
     # The android target depends on our own copy of openssl
-#    set(ANDROID_EXTRA_LIBS
- #       ${openssl_SOURCE_DIR}/arm/libcrypto_1_1.so
-  #      ${openssl_SOURCE_DIR}/arm/libssl_1_1.so
-   #     ${openssl_SOURCE_DIR}/arm64/libcrypto_1_1.so
-    #    ${openssl_SOURCE_DIR}/arm64/libssl_1_1.so
-     #   ${openssl_SOURCE_DIR}/x86/libcrypto_1_1.so
-      #  ${openssl_SOURCE_DIR}/x86/libssl_1_1.so
-       # ${openssl_SOURCE_DIR}/x86_64/libcrypto_1_1.so
-        #${openssl_SOURCE_DIR}/x86_64/libssl_1_1.so
-    #    CACHE INTERNAL ""
-     #   )
+    set(ANDROID_EXTRA_LIBS
+        ${openssl_SOURCE_DIR}/arm/libcrypto_1_1.so
+        ${openssl_SOURCE_DIR}/arm/libssl_1_1.so
+        ${openssl_SOURCE_DIR}/arm64/libcrypto_1_1.so
+        ${openssl_SOURCE_DIR}/arm64/libssl_1_1.so
+        ${openssl_SOURCE_DIR}/x86/libcrypto_1_1.so
+        ${openssl_SOURCE_DIR}/x86/libssl_1_1.so
+        ${openssl_SOURCE_DIR}/x86_64/libcrypto_1_1.so
+        ${openssl_SOURCE_DIR}/x86_64/libssl_1_1.so
+        CACHE INTERNAL ""
+        )
 
     # The android target is a library, not an executable
     add_library(${PROJECT_NAME} SHARED ${SOURCES} ${ANDROID_EXTRA_SOURCES})
