index a8a1588..eeda1fe 100644
--- a/node_modules/expo-barcode-scanner/android/build.gradle
+++ b/node_modules/expo-barcode-scanner/android/build.gradle
@@ -76,7 +76,6 @@ dependencies {
   unimodule "unimodules-core"
   unimodule 'expo-modules-core'
 
-  api 'com.google.android.gms:play-services-vision:19.0.0'
   api 'com.google.zxing:core:3.3.3'
 
   implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${safeExtGet('kotlinVersion', '1.4.21')}"
