diff --git a/app/build.gradle b/app/build.gradle
index 380becf..c74920a 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -53,6 +53,7 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
             buildConfigField "String", "dropbox_key", dropboxKey
             resValue 'string', "dropbox_key", dropboxKey
+            applicationIdSuffix '.release'
         }
     }
     packagingOptions {
@@ -63,11 +64,6 @@ android {
         exclude 'META-INF/DEPENDENCIES'
         exclude 'LICENSE.txt'
     }
-    productFlavors {
-        releaseflavor {
-            applicationId 'net.opendasharchive.openarchive.release'
-        }
-    }
 
     buildToolsVersion '28.0.3'
 }
